pypeit/core/findobj_skymask.py

```diff
@@ -147,7 +147,1159 @@ def create_skymask(sobjs, thismask, slit_left, slit_righ, box_rad_pix=None, trim
+    uni_obj = np.unique(sobjs_final.ECH_OBJID)
+    nobj_trim = uni_obj.size
+    spec_vec = np.arange(nspec)
+    specmid = nspec // 2
+
+    allmask = slitmask > -1
+    if inmask is None:
+        inmask = allmask
+
+    # Checks
+    if gd_orders.size != spec_min_max.shape[1]:
+        msgs.error("Number of good orders does not match the number of orders in spec_min_max")
+
+    # Loop over the objects one by one and adjust/predict the traces
+    pca_fits = np.zeros((nspec, norders, nobj_trim))
+
+    # Create the trc_inmask for iterative fitting below
+    trc_inmask = np.zeros((nspec, norders), dtype=bool)
+    for iord in range(norders):
+        trc_inmask[:,iord] = (
+            spec_vec >= spec_min_max[0,iord]) & (
+                spec_vec <= spec_min_max[1,iord])
+
+    for iobj in range(nobj_trim):
+        indx_obj_id = sobjs_final.ECH_OBJID == (iobj + 1)
+        # PCA predict all the orders now (where we have used the standard or slit boundary for the bad orders above)
+        msgs.info('Fitting echelle object finding PCA for object {:d}\{:d} with median SNR = {:5.3f}'.format(
+            iobj + 1,nobj_trim,np.median(sobjs_final[indx_obj_id].ech_snr)))
+        pca_fits[:,:,iobj] \
+                = tracepca.pca_trace_object(
+                    sobjs_final[indx_obj_id].TRACE_SPAT.T,
+                    order=coeff_npoly, npca=npca,
+                    pca_explained_var=pca_explained_var,
+                    trace_wgt=np.fmax(sobjs_final[indx_obj_id].ech_snr, 1.0)**2,
+                    debug=show_pca)
+
+        # Trial and error shows weighting by S/N instead of S/N^2 performs better
+        # JXP -- Updated to now be S/N**2, i.e. inverse variance, with fitting fit
+
+        # Perform iterative flux weighted centroiding using new PCA predictions
+        xinit_fweight = pca_fits[:,:,iobj].copy()
+        inmask_now = inmask & allmask
+        xfit_fweight = fit_trace(image, xinit_fweight, ncoeff, bpm=np.invert(inmask_now),
+                                 trace_bpm=np.invert(trc_inmask), fwhm=fwhm, maxdev=maxdev,
+                                 debug=show_fits)[0]
+
+        # Perform iterative Gaussian weighted centroiding
+        xinit_gweight = xfit_fweight.copy()
+        xfit_gweight = fit_trace(image, xinit_gweight, ncoeff, bpm=np.invert(inmask_now),
+                                 trace_bpm=np.invert(trc_inmask), weighting='gaussian', fwhm=fwhm,
+                                 maxdev=maxdev, debug=show_fits)[0]
+
+        #TODO  Assign the new traces. Only assign the orders that were not orginally detected and traced. If this works
+        # well, we will avoid doing all of the iter_tracefits above to make the code faster.
+        for iord, spec in enumerate(sobjs_final[indx_obj_id]):
+            # JFH added the condition on ech_frac_was_fit with S/N cut on 7-7-19.
+            # TODO is this robust against half the order being masked?
+            if spec.ech_frac_was_fit & (spec.ech_snr > 1.0):
```

Comment:
Don't use `&` with single booleans.  Instead: 

`if spec.ech_frac_was_fit and spec.ech_snr > 1:`

Reply:
modified